C/CvxCompress/build_tarballs.jl

```diff
@@ -3,35 +3,45 @@
 using BinaryBuilder, Pkg
 
 name = "CvxCompress"
-version = v"1.1.0"
+version = v"1.2.0"
 
 # Collection of sources required to build CvxCompress
 sources = [
     GitSource(
         "https://github.com/ChevronETC/CvxCompress.git",
-        "04dc59a4fab76ac612580ac69a9266e15db4fe17"
+        "132984d055be77cb166fe2e75dc77f4fd082e544"
     ),
     DirectorySource("./bundled"),
 ]
 
 # Bash recipe for building across platforms
 script = raw"""
+install_license ${WORKSPACE}/srcdir/CvxCompress/LICENSE.md
+
 cd ${WORKSPACE}/srcdir/CvxCompress
 atomic_patch -p1 ${WORKSPACE}/srcdir/patches/cross-compile-cvxcompress-gencode.patch
+# Pull simde
+git submodule update --init
+
 ${CXX_FOR_BUILD} -O2 -o CvxCompress_GenCode Wavelet_Transform_Slow.cpp CvxCompress_GenCode.cpp
 ./CvxCompress_GenCode
-FLAGS=()
-if [[ "${target}" == *-apple-* ]]; then
-    FLAGS=(LDFLAGS="-fopenmp -lm")
+
+if [[ "${target}" == *-freebsd* ]] || [[ "${target}" == *-apple-* ]]; then
+    CC=gcc
+    CXX=g++
```

Comment:
Because that's what the documentation recommends. 

https://docs.binarybuilder.org/stable/build_tips/#Using-GCC-on-macOS-and-FreeBSD

Reply:
To be clear, the question was meant to be "why forcing gcc at all?  what's the problem with using the default compiler (clang on these platforms)?"

Reply:
The default clang at least on apple doesn't support openmp so gcc is a better choice

Reply:
> The default clang at least on apple doesn't support openmp so gcc is a better choice

That's not accurate, you only need to explicitly bring an openmp runtime, like llvmopenmp :slightly_smiling_face: https://github.com/JuliaPackaging/Yggdrasil/blob/2d4f71c0595eba5e2e6fbebb25ecb82d80f05ce9/S/SZ/build_tarballs.jl#L70-L71

Reply:
`llvmopenmp` creates quite annoying complications in many projects that use this library, as it conflicts with the system llvm used for code generation, so I'd rather avoid it.

Reply:
`LLVMOpenMP_jll` will be loaded into the load path when this package is used and take precedence over the system library path. It's possible to avoid it but it's preferable to keep this with minimal dependencies.

Reply:
I mean, the literally same applies to libgomp and you aren't complaining about that one.  And the whole point of using JLLs is to use system libraries, yes, to have a consistent environment on all platforms.

Reply:
Also, the probability of having llvm openmp already installed on the system is very small, compared to libgomp.

Reply:
> What's the fundamental issue with using gcc?

That's not the compiler used by default on these systems, which makes it a less "natural" choice.

Reply:
Considering that the proposed change doesn't work and that documentation explicitly state that using GCC on macos and BSD for compilation is even recommended in some case I really don't see why a sane and working GCC build wouldn't be acceptable.

Reply:
> Considering that the proposed change doesn't work

If you're referring to
```
[19:30:22] In file included from Block_Copy.cpp:1:
[19:30:22] In file included from ./Block_Copy.hxx:6:
[19:30:22] In file included from ./simde/x86/avx512.h:30:
[19:30:22] ./simde/x86/avx512/types.h:634:27: error: typedef redefinition with different types ('simde__m128bh' (vector of 4 'simde_float32' values) vs '__attribute__((__vector_size__(8 * sizeof(__bf16)))) __bf16' (vector of 8 '__bf16' values))
[19:30:22]   634 |     typedef simde__m128bh __m128bh;
[19:30:22]       |                           ^
[19:30:22] /opt/x86_64-linux-musl/lib/clang/18/include/emmintrin.h:48:16: note: previous definition is here
[19:30:22]    48 | typedef __bf16 __m128bh __attribute__((__vector_size__(16), __aligned__(16)));
[19:30:22]       |                ^
[19:30:22] 1 error generated.
```
since I get similar errors if I try to compile the code locally outside of BinaryBuilder with clang-18 and clang-19 on x86_64, I have the suspicion the error is in the source code, not the compiler.

Reply:
The source code is highly optimized intrinsic and will not be modified this merely add support for ARM through simde but there will not be work done on the source. 
The build is stable and valid with gcc so this should be merged.

Reply:
> The source code is highly optimized intrinsic and will not be modified this merely add support for ARM through simde but there will not be work done on the source.

In case it wasn't obvious, I believe the bug is in simde, since the error above is telling you that [simde defines a type](https://github.com/simd-everywhere/simde/blob/9cbeced99f0b6a326dd467f3468f3eabb40534b7/simde/x86/avx512/types.h#L634) which conflicts with a [type defined by clang header files](https://github.com/llvm/llvm-project/blob/b46a0482f9e4c0ee82b38da794b20f8f1a76f044/clang/lib/Headers/emmintrin.h#L48).  Locally outside of BinaryBuilder I got a different error but still about a conflict between another type defined by simde and a type defined by clang.  I suspect they've got some guards wrong.  But I trashed the directory where I did that test and I don't bother trying it again to bring the error I got.  I should say that I find it surprising that clang defines bfloat16 types in the SSE header file, but they moved them there in https://github.com/llvm/llvm-project/commit/e0fb01e97b6b7d2fe66b17b36eeb98aa78c6e3bb, hopefully with a good reason (the comment says "Both _Float16 and __bf16 require SSE2 being enabled." but doesn't elaborate much more than that why it makes sense to put them there rather than the avx512-bf16-specific header file)

That said, I don't have much more energy arguing that this isn't ideal and should be fixed upstream rather than ignored.